arrows-es6/index.html

```diff
@@ -0,0 +1,85 @@
+---
+feature_name: ES6 Arrow functions
+chrome_version: 45
+feature_id: 5047308127305728
+additional_head_content: <meta name="optional" value="optional">
+---
+
+<h3>Background</h3>
+
+<p>ES6 fat arrow functions have a shorter syntax compared to
+function expressions and lexically bind the <code>this</code> value. Arrow functions are always anonymous and effectively turn
+<code>function (arguments) { expression }</code> into arguments => expression. If using an expression after an arrow, the return is implicit, so no <code>return</code> is required.</p>
+
+{% capture initial_output_content %}
+<p>This can be used to set initial sample output.</p>
+{% endcapture %}
+{% include output_helper.html initial_output_content=initial_output_content %}
+
+{% capture js %}
+'use strict';
+// Filter an array for only odd numbers
+let numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+// Before...
+let es5_odd_numbers = numbers.filter(function(number) {
+  return number % 2;
+});
+ChromeSamples.log(es5_odd_numbers);
+
+// After...
+let es6_odd_numbers = numbers.filter(number => number % 2);
+ChromeSamples.log(es6_odd_numbers);
+
+// Parens are optional depending on the number of arguments:
+let square = x => x * x;
+ChromeSamples.log(square(10));
+
+let add = (a, b) => a + b;
+ChromeSamples.log(add(3,4));
+
+// `return` is implied if using an expression after an arrow.
+let developers = [{ name: 'Rob' }, { name: 'Jake' }];
+// Before...
+let es5_output = developers.map(function(developer) { return developer.name; });
+ChromeSamples.log(es5_output);
+
+// After...
+let es6_output = developers.map(developer => developer.name);
+ChromeSamples.log(es6_output);
+
+// Fat arrows change how `this` is handled.
+
+// Before...
+// In ES5, `bind()` or var that = this; are necessary as functions
+// create their own `this`. We need to store the parent `this` in
+// a variable that can be referenced in the callback or take care
+// of binding ourselves.
+var counter_es5 = function() {
+  this.seconds = 0;
+  var that = this;
+  setInterval(function() {
+    that.seconds++;
+  }, 1000); // or }.bind(this), 1000) and skip that = this
+}
+
+var counterA = new counter_es5();
+setTimeout(function() {
+  ChromeSamples.log(counterA.seconds);
+}, 1200);
+
+// After...
+// ES6 Arrows instead bind `this` to the immediate enclosing
+// lexical scope:
+let counter_es6 = function() {
+  this.seconds = 0;
+  setInterval(() => this.seconds++, 1000);
+}
+
+let counterB = new counter_es6();
+setTimeout(() => ChromeSamples.log(counterB.seconds), 1200);
+
+{% endcapture %}
+{% include js_snippet.html js=js %}
+
+{% include css_snippet.html css=css %}
```

Comment:
You can leave out `{% include css_snippet.html css=css %}`. It's just needed if there is inline CSS that you're demonstrating as part of the sample.